deployer/strategies/main.go

```diff
@@ -0,0 +1,89 @@
+package strategy
+
+import (
+	"fmt"
+	"github.com/flynn/flynn/controller/client"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/deployer/types"
+	"time"
+)
+
+type Performer interface {
+	Perform(*deployer.Deployment, chan<- deployer.DeploymentEvent) error
+}
+
+func GetStrategy(strategy string, client *controller.Client) (Performer, error) {
```

Comment:
Since the strategies don't really need any state, this could probably just return a function, e.g.:

```go
package strategy

type PerformFunc func(client *controller.Client, *deployer.Deployment, chan<- deployer.DeploymentEvent) error

var performFuncs = map[string]PerformFunc{
	"all-at-once": allAtOnce,
	"one-by-one": oneByOne,
}

func Get(strategy string) (PerformFunc, error) {
	if f, ok := performFuncs[strategy]; ok {
		return f, nil
	}
	return nil, fmt.Errorf("Unknown strategy '%s'!", strategy)
}

func allAtOnce(client *controller.Client, *deployer.Deployment, chan<- deployer.DeploymentEvent) error {}
func oneByOne(client *controller.Client, *deployer.Deployment, chan<- deployer.DeploymentEvent) error {}
```